cuML/src/random_projection/utils.hxx

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2018, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "rproj_c.h"
+#include <random/rng.h>
+
+using namespace MLCommon::Random;
+
+inline void sample_without_replacement(size_t n_population, size_t n_samples,
```

Comment:
We are hopefully going to have a sampling w/o replacement prim on device soon. Was discussing this with @vishalmehta1991 earlier today.